### PR TITLE
Dom

### PR DESCRIPTION
**Pull request by rahmanoff**
### DOM

[Demo](https://rahmanoff.github.io/js-dom/) |
[Code base](https://github.com/rahmanoff/js-dom)

The code is submitted in a dedicated feature branch.

Only code files are submitted.

Please, review.

### CODE REVIEW

**Comment by rahmanoff** [comment]
Hi!
JS changed due to Code Review Rules.
Pls review.

File: submissions/Kostiantyn_Rakhmanov/dom/index.html [hunk 1]
```diff
@@ -0,0 +1,66 @@
...
+        <footer class="footer">&copy; 2022 Most Expensive Alfa Romeo cars by
+            <a class="footer__link" href="https://github.com/rahmanoff">rahmanoff</a>
+        </footer>
+    </div>
+
+
+</body>
+<script src="./main.js"></script>
+
+</html>
```
Comment (madmaxWMFU) [inline submissions/Kostiantyn_Rakhmanov/dom/index.html:66]: Add an empty line.

File: submissions/Kostiantyn_Rakhmanov/dom/main.js [hunk 2]
```diff
@@ -0,0 +1,41 @@
...
+    item = alfas[itemId]
+        section.innerHTML = `
+        <img class="article__img" src="${item.img}" alt="Alfa Romeo">
+        <h2 class="article__title" id="article__title">${item.model}</h2>
+        <p class="article__price">${item.price}</p>
+        <article class="article__text" id="article__text">
+        <p>${item.description}</p>
+        </article>
+        `;
+});
```
Comment (madmaxWMFU) [inline submissions/Kostiantyn_Rakhmanov/dom/main.js:41]: Also, the empty line.

File: submissions/Kostiantyn_Rakhmanov/dom/main.js [hunk 3]
```diff
@@ -0,0 +1,41 @@
...
+    
+];
+
+const menu = document.querySelector('.menu__items');
+const section = document.querySelector('.section');
+
+
+menu.addEventListener("click", ({ target }) => {
+    const itemId = target.parentElement.id;
+    item = alfas[itemId]
```
Comment (madmaxWMFU) [inline submissions/Kostiantyn_Rakhmanov/dom/main.js:32]: You used destruction it is good.
Then you can try it in this part of the code.
Recommend to use the method of Array: `find` or `filter`

**Review by madmaxWMFU — CHANGES_REQUESTED** [review]
Good work see some comments below

File: submissions/Kostiantyn_Rakhmanov/dom/index.html (hunk 1 above)
Comment (rahmanoff) [reply submissions/Kostiantyn_Rakhmanov/dom/index.html]: Hi!
Done.

File: submissions/Kostiantyn_Rakhmanov/dom/main.js (hunk 2 above)
Comment (rahmanoff) [reply submissions/Kostiantyn_Rakhmanov/dom/main.js]: Hi!
Done

**Comment by rahmanoff** [comment]
Hi!
Pls re-review!

File: submissions/Kostiantyn_Rakhmanov/dom/styles.css [hunk 4]
```diff
@@ -0,0 +1,117 @@
...
+    margin: 10px auto;
+    width: 100%;
+    max-height: 450px;
+    object-fit: contain;
+}
+
+.footer {
+    text-align: center;
+    font-size: 12px;
+}
```
Comment (madmaxWMFU) [inline submissions/Kostiantyn_Rakhmanov/dom/styles.css:117]: Also, the empty line.

File: submissions/Kostiantyn_Rakhmanov/dom/main.js [hunk 5]
```diff
@@ -0,0 +1,41 @@
...
+    let alfa = alfas.filter(item => (item.id == itemId))
+    section.innerHTML = `
+            <img class="article__img" src="${alfa[0].img}" alt="Alfa Romeo">
+            <h2 class="article__title" id="article__title">${alfa[0].model}</h2>
+            <p class="article__price">${alfa[0].price}</p>
+            <article class="article__text" id="article__text">
+            <p>${alfa[0].description}</p>
+            </article>
+            `;
+});
```
Comment (madmaxWMFU) [inline submissions/Kostiantyn_Rakhmanov/dom/main.js:41]: Add an empty line.

File: submissions/Kostiantyn_Rakhmanov/dom/main.js [hunk 6]
```diff
@@ -0,0 +1,41 @@
...
+    
+];
+
+const menu = document.querySelector('.menu__items');
+const section = document.querySelector('.section');
+
+
+menu.addEventListener("click", ({ target }) => {
+    const itemId = target.parentElement.id;
+    let alfa = alfas.filter(item => (item.id == itemId))
```
Comment (madmaxWMFU) [inline submissions/Kostiantyn_Rakhmanov/dom/main.js:32]: Must be a `const`

File: submissions/Kostiantyn_Rakhmanov/dom/main.js [hunk 7]
```diff
@@ -0,0 +1,41 @@
...
+    section.innerHTML = `
+            <img class="article__img" src="${alfa[0].img}" alt="Alfa Romeo">
+            <h2 class="article__title" id="article__title">${alfa[0].model}</h2>
+            <p class="article__price">${alfa[0].price}</p>
+            <article class="article__text" id="article__text">
+            <p>${alfa[0].description}</p>
+            </article>
+            `;
```
Comment (madmaxWMFU) [inline submissions/Kostiantyn_Rakhmanov/dom/main.js:40]: Read about [destruction](https://javascript.info/destructuring-assignment)

**Review by madmaxWMFU — CHANGES_REQUESTED** [review]

File: submissions/Kostiantyn_Rakhmanov/dom/styles.css (hunk 4 above)
Comment (rahmanoff) [reply submissions/Kostiantyn_Rakhmanov/dom/styles.css:117]: Hi!
Sorry about these lines. Looks like VSC deleled all my additional lines. Will check it in future.

File: submissions/Kostiantyn_Rakhmanov/dom/main.js [hunk 8]
```diff
@@ -0,0 +1,42 @@
...
+];
+
+const menu = document.querySelector('.menu__items');
+const section = document.querySelector('.section');
+
+
+menu.addEventListener("click", ({ target }) => {
+    const itemId = target.parentElement.id;
+    const alfa = (alfas.filter(item => (item.id == itemId)));
+    const { 0: {img, model, price, description} } = alfa;
```
Comment (madmaxWMFU) [inline submissions/Kostiantyn_Rakhmanov/dom/main.js:33]:
```suggestion
    const alfa = alfas.filter((id) => id == itemId);
    const [ {img, model, price, description} ] = alfa;

```

**Review by madmaxWMFU — CHANGES_REQUESTED** [review]

File: submissions/Kostiantyn_Rakhmanov/dom/main.js (hunk 8 above)
Comment (rahmanoff) [reply submissions/Kostiantyn_Rakhmanov/dom/main.js:33]: Hi!
Looks clear, but it is not working.
In your case, alfa = [], empty Array.

Comment (madmaxWMFU) [reply submissions/Kostiantyn_Rakhmanov/dom/main.js:33]: Sorry, had a mistake. Try this.
```
    const alfa = alfas.filter(({ id }) => id == itemId);
    const [ {img, model, price, description} ] = alfa;
```

Comment (rahmanoff) [reply submissions/Kostiantyn_Rakhmanov/dom/main.js]: Now it is working.
Thanks for suppport!

**Review by madmaxWMFU — APPROVED** [review]
Great, I approve of your work. Good luck with future tasks!

**Pull request merged**